Code/GraphMol/CIPLabeler/Node.h

```diff
@@ -0,0 +1,125 @@
+//
+//
+//  Copyright (C) 2020 Schrödinger, LLC
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+#pragma once
+
+#include <vector>
+
+#include "Descriptor.hpp"
+#include "Mancude.h"
+#include "Edge.h"
+
+namespace RDKit {
+
+class Atom;
+
+namespace CIPLabeler {
+
+class Digraph;
+
+class Node {
+public:
+  /**
+   * Flag indicates whether the node has been expanded.
+   */
+  static const int EXPANDED = 0x1;
+
+  /**
+   * Flag indicates whether the node was duplicated
+   * at a ring closure.
+   */
+  static const int RING_DUPLICATE = 0x2;
+
+  /**
+   * Flag indicates whether the node was duplicated
+   * at a bond with order &gt; 1.
+   */
+  static const int BOND_DUPLICATE = 0x4;
+
+  /**
+   * Mask to check if a node is duplicated.
+   */
+  static const int DUPLICATE = 0x6;
+
+  /**
+   * Node was created for an implicit hydrogen,
+   * the 'atom' value will be null.
+   */
+  static const int IMPL_HYDROGEN = 0x8;
+
+private:
+  Digraph *g;
+  Atom *atom;
+  int dist;
+  Fraction d_atomic_num;
+  Descriptor aux = Descriptor::NONE;
```

Comment:
Should it start at unknown?

Reply:
I think that's not possible: my intuition from the code I have seen is that `UNKNOWN` means that the label cannot be determined using the rule set currently being applied.

In the Java version, there isn't a `NONE` in the enum, so that `aux` is just left un initialized, and is compared to `Null` (I think I recall), which we can't do in C++. Since there are checks for `Null` and also for `UNKNOWN`, I am not sure what would happen if we just replace `NONE` with `UNKNOWN`.